laurent22/joplin
# Android: Plugins: Autohide the plugin panel toggle in toolbar to increase size for notebook dropdown

# Summary

This pull request does the following:
- Autohides the plugin panel toggle in the mobile toolbar to increase the space available for the notebook dropdown.
- Refactors `Dropdown.tsx` to:
   1. Update the list size when the container size changes.
   2. Use a single `this.onCloseList` callback, rather than repeatedly wrapping a `closeList()` callback as `() => { closeList() }` or similar.
   3. Support including children to the right of the toggle button that will be hidden when the dropdown list is shown (to support autohiding the panel toggle button).

This allows the note list to have the same amount of space available as before https://github.com/laurent22/joplin/pull/10180.

# Notes

- Currently, this is done by making the plugin button a child of the dropdown. The dropdown then includes the child's size in its calculations for the size of the dropdown list.
- A previous attempt kept the plugin button in its prior location, but hid it when the notebook list was shown. This worked, but caused the notebook list to resize just after opening (the notebook list would very briefly be a smaller size).
    - Many of the refactoring changes come from this attempt. 

# Screenshots

![screenshot: Joplin mobile note editor running on a device with a small screen. The plugin icon is visible and the dropdown is closed](https://github.com/laurent22/joplin/assets/46334387/6dc99fb0-a623-48d1-b1b2-53961eabcc2d)![screenshot: Same note editor as in the previous image, except now the plugin icon is hidden and the notebook selector dropdown is visible](https://github.com/laurent22/joplin/assets/46334387/381ad964-e6b6-4a2e-93a4-8c139bc0096c)

# Testing

While this pull request includes an automated test that checks part of its functionality, its use in the header bar must be tested manually. To do this,
1. Install a plugin that registers a panel (e.g. the History Panel plugin)
1. Open the note list and long-press on a note
2. Click "move to notebook" and verify that the plugin icon is hidden
3. Close the "move to notebook" dropdown and verify that the plugin icon is visible
4. Open a note
5. Open the notebook selector dropdown and verify that the plugin icon is hidden.
6. Close the notebook selector dropdown and verify that the plugin icon is visible again.

This has been tested successfully on Android 13.
